### DISCUSSION TITLE
Concourse build log to splunk

### DISCUSSION THREAD

**Discussion by ranjithfd** [concourse/concourse#5685] [General]
Hi Team,

 Is there any way to send concourse build log to splunk. If yes, please share the procedure.

REgards,
Ranjith D

**Comment by jamieklassen** [comment]
There is a Concourse subcomponent called the 'Syslog drainer' that is responsible for asynchronously forwarding the logs from finished builds to an external log store via the syslog protocol. I am not a splunk expert, but I believe it can be configured to listen for syslog messages.

On the concourse side, you can configure the binary to target a given syslog endpoint using flags or environment variables as described here:
https://github.com/concourse/concourse/blob/72c7cdad6f2ef05036c6c9fb44a2fc5eee6cbb88/atc/atccmd/command.go#L224-L230

This configuration is also exposed via the helm chart:
https://github.com/concourse/concourse-chart/blob/bfee5dc26ce2713eac925517e41c1e8d1cd8a15b/values.yaml#L747-L772

and the bosh release:
https://github.com/concourse/concourse-bosh-release/blob/5336342ac6d22f8ab79b0141c9177402db3b4f4e/jobs/web/spec#L1430-L1455

Our docs website lacks an adequate description of this feature. I could see some benefit in adding a 'Logging' section under https://concourse-ci.org/operation.html and/or a 'Syslog drainer' sub-subsection under https://concourse-ci.org/internals.html#component-atc.
